main.go

```diff
@@ -73,85 +78,33 @@ func main() {
 
 	printVersion()
 
-	namespace, err := getWatchNamespace()
+	_, err := getWatchNamespace()
 	if err != nil {
 		log.Error(err, "unable to get WatchNamespace; unable to start manager")
 		os.Exit(1)
 	}
-	log.Info(fmt.Sprintf("Namespace: %s", namespace))
-
-	// Get a config to talk to the apiserver
-	cfg, err := config.GetConfig()
-	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	ctx := context.TODO()
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "updateservice-operator-lock")
-	if err != nil {
-		log.Error(err, "Could not become leader")
-		os.Exit(1)
-	}
-
-	// Set default manager options
-	options := manager.Options{
-		Namespace:          namespace,
-		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
-	}
-
-	// Add support for MultiNamespace set in WATCH_NAMESPACE (e.g ns1,ns2)
-	// Note that this is not intended to be used for excluding namespaces, this is better done via a Predicate
-	// Also note that you may face performance issues when using this with a high number of namespaces.
-	// More Info: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder
-	if strings.Contains(namespace, ",") {
-		options.Namespace = ""
-		options.NewCache = cache.MultiNamespacedCacheBuilder(strings.Split(namespace, ","))
-	}
 
-	// Create a new manager to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, options)
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		Port:               9443,
+		LeaderElection:     enableLeaderElection,
+		LeaderElectionID:   "2b19ccee.openshift.io",
+	})
 	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	log.Info("Registering Components.")
-
-	// Setup Scheme for all resources
-	if err := updateservicev1.AddToScheme(mgr.GetScheme()); err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	if err := configv1.Install(mgr.GetScheme()); err != nil {
-		log.Error(err, "")
+		log.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	if err := routev1.Install(mgr.GetScheme()); err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	opts := uscontroller.Options{
-		OperandImage: os.Getenv("OPERAND_IMAGE"),
-	}
-	if opts.OperandImage == "" {
-		log.Error(errors.New("Must set envvar OPERAND_IMAGE"), "")
-		os.Exit(1)
-	}
-	// Usually this is done with an init() function in the package, but since we
-	// need to provide input, we're doing it here instead.
-	uscontroller.AddToManagerFuncs = append(uscontroller.AddToManagerFuncs, func(mgr manager.Manager) error {
-		return uscontroller.Add(mgr, opts)
-	})
-	// Setup UpdateService controllers
-	if err := uscontroller.AddToManager(mgr); err != nil {
-		log.Error(err, "")
+	if err = (&controllers.UpdateServiceReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("UpdateService"),
+		Scheme: mgr.GetScheme(),
```

Comment:
[e2e][1] has lots of logs like:

```
2020-11-12T22:38:21.734Z	ERROR	controller_updateservice	CreateDeploymentFailed	{"Request.Namespace": "openshift-updateservice", "Request.Name": "example", "error": "Deployment.apps \"example\" is invalid: [spec.template.spec.containers[0].image: Required value, spec.template.spec.containers[1].image: Required value]"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/src/github.com/openshift/cincinnati-operator/vendor/github.com/go-logr/zapr/zapr.go:132
github.com/openshift/cincinnati-operator/controllers.handleErr
	/go/src/github.com/openshift/cincinnati-operator/controllers/updateservice_controller.go:172
github.com/openshift/cincinnati-operator/controllers.(*UpdateServiceReconciler).ensureDeployment
```

To pass in an image you need `OPERAND_IMAGE` handling like [this][2].

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cincinnati-operator/82/pull-ci-openshift-cincinnati-operator-master-operator-e2e/1327005667935916032
[2]: https://github.com/openshift/cincinnati-operator/pull/76/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R85

Reply:
I've pushed 7122c12 to address this; squash it in if it looks good to you.

Reply:
done